hoomd/md/many_body.py

```diff
@@ -0,0 +1,343 @@
+import hoomd
+from hoomd.md import _md
+from hoomd.md.pair import Pair
+from hoomd.data.parameterdicts import TypeParameterDict
+from hoomd.data.typeparam import TypeParameter
+
+
+class Triplet(Pair):
+    """Common three body potential documentation.
+
+    Users should not invoke :py:class:`Triplet` directly. It is a base command
+    that provides common features to all standard triplet forces. Common
+    documentation for all three-body potentials is documented here.
+
+    Triplet potentials work similar to pair potentials in that they utilize
+    turn-on and cutoff distances :math:`r_{\\mathrm{on}}` and :math:`r_{\\mathrm{cut}}`,
+    respectively, along with energy shifting and smoothing modes. The detailed
+    documentation for this is located in :py:class:`hoomd.md.pair.Pair`.
+
+    Warning:
+        Currently HOOMD does not support reverse force communication between MPI
+        domains on the GPU. Since reverse force communication is required for
+        the calculation of three-body potentials, attempting to use this
+        potential on the GPU with MPI will result in an error.
+    """
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+    def _attach(self):
+        super()._attach()
+        self.nlist._cpp_obj.setStorageMode(
+            _md.NeighborList.storageMode.full)
+
+
+class Tersoff(Triplet):
+    """Tersoff Potential.
+
+    Args:
+        nlist (:py:mod:`hoomd.md.nlist`): Neighbor list
+        r_cut (float): Default cutoff radius (in distance units).
+        r_on (float): Default turn-on radius (in distance units).
+        mode (str): Energy shifting mode.
```

Comment:
I might be missing something but I don't see the definition of `mode` or the allowed options for `mode`. If I've missed that, you can just resolve this conversation.

Reply:
The definition of mode is in `md.pair.Pair`. It is explained in `md.many_body.Triplet`, that the triplet potentials use many of the same arguments as the pair potentials, and the common documentation on those is in `md.pair.Pair`. Do you think this is enough, or should we be more explicit about this somehow?

Reply:
Yes, we should be more explicit. We haven't officially decided on a policy for this but the practice we last discussed was to use the :inherited-members: sphinx option for classes like `Triplet` so that the relevant inherited docs are present in the local HTML page as the classes users actually use. See `module-hommd-operation.rst` for an example of now to enable this option for one class and not others.

Reply:
What we need to be showing is the documentation under the class declaration, not the inherited members, and I can't find an option for this. Any ideas?

Reply:
Sphinx doesn't inherit docstrings themselves, so you'll have to copy and paste.